docs/reusing-schemas.md

```diff
@@ -0,0 +1,246 @@
+
+  The common definitions (data structures) are not specific to any controller or
+  business domain logic.
+  """
+  def create_swagger_definitions(%{} = schemas) do
+    Map.merge(
+      %{
+        Error: 
+          swagger_schema do
+            properties do
+              code(:string, "Error code", required: true)
+              message(:string, "Error message", required: true)
+            end
+          end,
+        Errors:
+          swagger_schema do
+              properties do
+                errors(
+                  Schema.new do
+                    title("Errors")
+                    description("A collection of Errors")
+                    type(:array)
+                    items(Schema.ref(:Error))
+                  end
+                )
+              end
+          end          
+      },
+      schemas
+    )
+  end
+end
+```
+
+As you can see, it is also possible to reference other common schemas defined inside the `create_swagger_definitions/1`.
+
+## Reusing the common schemas
+
+Now, instead of defining the `Error` schema in every controller, we can just use `create_swagger_definitions/1`
+inside our controllers.
+
+```elixir
+defmodule ProjectsController do
+  import CommonSchemas, only: [create_swagger_definitions: 1]
+
+  use HelloWeb, :controller
+  use PhoenixSwagger
+
+  swagger_path :index do
+    get "/projects"
+    produces "application/json"
+    parameter("Authorization", :header, :string, "OAuth2 access token", required: true)
+    parameters do      
+      sort_by :query, :string, "The property to sort by"
+      sort_direction :query, :string, "The sort direction", enum: [:asc, :desc], default: :asc
+      company_id :string, :query, "The company id"
+    end
+    response(200, "OK", Schema.ref(:ListOfProjects))
+    response(401, "Unauthorized", Schema.ref(:Error))
+  end
+  
+  
+
+  @doc false
+  def swagger_definitions do
+    create_swagger_definitions(%{
+        ListOfProjects: ...schema definition...
+    })  
+  end
+end
+```
+
+```elixir
+defmodule BooksController do
+  import CommonSchemas, only: [create_swagger_definitions: 1]
+
+  use HelloWeb, :controller
+  use PhoenixSwagger
+  
+  swagger_path :index do
+    get "/books"
+    produces "application/json"
+    parameter("Authorization", :header, :string, "OAuth2 access token", required: true)
+    parameters do      
+      sort_by :query, :string, "The property to sort by"
+      sort_direction :query, :string, "The sort direction", enum: [:asc, :desc], default: :asc
+      company_id :string, :query, "The company id"
+    end
+    response(200, "OK", Schema.ref(:ListOfBooks))
+    response(401, "Unauthorized", Schema.ref(:Error))
+  end
+  
+  
+
+  @doc false
+  def swagger_definitions do
+    create_swagger_definitions(%{
+        ListOfBooks: ...schema definition...
+    }) 
+  end
+end
+```
+
+To avoid importing `create_swagger_definitions/1` in every controller, find a `HelloWeb` module and add an import
```

Comment:
This advice feels a bit opinionated. Users are free to organise their projects and code as they like. My preference is to delete this file for API projects trading off some dry-ness for explicitness of imports.

Reply:
Yes I agree, this is my opinionated hint how to use something that pre-generated phoenix project already contains. Should we remove this section then ?